docs/sources/operations/storage/retention.md

```diff
@@ -6,13 +6,14 @@ weight:  600
 ---
 # Log retention
 
-Retention in Grafana Loki is achieved either through the [Table Manager](#table-manager) or the [Compactor](#compactor).
+Retention in Grafana Loki is achieved through the [Compactor](#compactor).
+It is the recommended way of applying retention for [boltdb-shipper]({{< relref "./boltdb-shipper" >}}) and [tsdb]({{< relref "./tsdb" >}}).
 
-By default, when `table_manager.retention_deletes_enabled` or `compactor.retention_enabled` flags are not set, then logs sent to Loki live forever.
+Though [Table Manager]({{< relref "./table-manager" >}}) can perform retention on [boltdb-shipper]({{< relref "./boltdb-shipper" >}}) and other [legacy index types]({{< relref "../../storage#index-storage" >}}), it is now deprecated.
 
-Retention through the [Table Manager]({{< relref "./table-manager" >}}) is achieved by relying on the object store TTL feature, and will work for both [boltdb-shipper]({{< relref "./boltdb-shipper" >}}) store and chunk/index store. However retention through the [Compactor]({{< relref "./boltdb-shipper#compactor" >}}) is supported only with the [boltdb-shipper]({{< relref "./boltdb-shipper" >}}) and tsdb store.
+By default `compactor.retention_enabled` flag is not set, then logs sent to Loki live forever.
```

Comment:
I always struggle to write about the usage of a configuration setting. Personally I use "flag" in the context of "CLI flag" and "configuration" in the context of the YAML config. However, I don't know if there is a convention.

That said, I would interpret `compactor.retention_enabled` as CLI flag and would use it like so: `-compactor.retention_enabled=true` however, the flag is actually with a dash, rather than an underscore.

Maybe this is nitpicking, but I think we should find a way to be consistent across the documentation.

Reply:
> however, the flag is actually with a dash, rather than an underscore.
thanks for catching this, I'll correct it.

agree that we need to be consistent.
I prefer the CLI flag since it's easier to refer to nested YAML config options with it. wdyt?

Reply:
Sounds good to me.